timelock-impl/src/main/java/com/palantir/atlasdb/timelock/clock/ClockSkewComparer.java

```diff
@@ -46,12 +45,11 @@ public ClockSkewComparer(String server, ClockSkewEvents events, RequestTime prev
     }
 
     public void compare() {
-        Preconditions.checkArgument(maxElapsedTime > 0,
-                "A positive maxElapsedTime is expected");
-        Preconditions.checkArgument(minElapsedTime > 0,
-                "A positive minElapsedTime is expected");
-        Preconditions.checkArgument(remoteElapsedTime > 0,
-                "A positive remoteElapsedTime is expected");
+        if (clockHasNotMovedForwards()) {
+            // The clock not moving forwards is already tracked by the ReversalDetectingClockService, so it is fine
+            // to no op here. We don't want to use these values.
+            return;
+        }
 
         if (hasTooMuchTimeElapsedSincePreviousRequest()) {
```

Comment:
i wonder if we should get rid of these two checks (`hasTooMuchTimeElapsedSincePreviousRequest` and `requestsTookTooLongToComplete `). even if a lot of time has elapsed, or the request took a while to return, i think we're still interested in knowing if there was any clock skew; we just ALSO want to know the period over which the skew occurred and how long the request took.

Reply:
Yep, that's fine. I think one caveat though is that I'd rather not log at `ERROR` in those cases, since as we move to a more operational model this might cause alerts.

I've moved it to be:

- ERROR if it exceeds the error threshold *and* passes the checks
- WARN if it exceeds the warn threshold and is not logged at ERROR.